tests/testthat/test-build-news.R

```diff
@@ -1,18 +1,81 @@
+test_that("data_news works as expected - h1", {
+  skip_if_no_pandoc()
+
+  temp_pkg <- list(
+    src_path = withr::local_tempdir(pattern = "pkgdown-news"),
+    bs_version = 5
+  )
+
+  write_lines(
+    c(
+      "# testpackage 1.0.0.9000", "",
+      "* bullet (#222 @someone)", "",
+      "# testpackage 1.0.0", "",
+      "## sub-heading", "",
+      "* first thing (#111 @githubuser)", "",
+      "* second thing", ""
+    ),
+    file.path(temp_pkg$src_path, "NEWS.md")
+  )
+
+  expect_snapshot(data_news(temp_pkg))
+})
+
+test_that("data_news works as expected - h2", {
+  skip_if_no_pandoc()
+
+  temp_pkg <- list(
+    src_path = withr::local_tempdir(pattern = "pkgdown-news"),
+    bs_version = 5
+  )
+
+  write_lines(
+    c(
+      "## Changes in v1.0.0.9000", "",
+      "* bullet (#222 @someone)", "",
+      "## Changes in v1.0.0", "",
+      "### sub-heading", "",
+      "* first thing (#111 @githubuser)", "",
+      "* second thing", ""
+    ),
+    file.path(temp_pkg$src_path, "NEWS.md")
+  )
+
+  expect_snapshot(data_news(temp_pkg))
+})
+
 test_that("github links are added to news items", {
   skip_if_no_pandoc()
 
-  pkg <- as_pkgdown(
-    test_path("assets/news-github-links"),
```

Comment:
I removed that assets folder. for assets/news sadly it's used in other tests (e.g. for search index building) so it'll require more test refactoring.